homeassistant/components/meteo_france/sensor.py

```diff
@@ -1,168 +1,231 @@
-            self._state = None
+        """Return the device class."""
+        return SENSOR_TYPES[self._type][ENTITY_CLASS]
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added to the entity registry."""
+        return SENSOR_TYPES[self._type][ENTITY_ENABLE]
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {ATTR_ATTRIBUTION: ATTRIBUTION}
+
+    @property
+    def available(self):
+        """Return if state is available."""
+        return self.coordinator.last_update_success
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed."""
+        return False
+
+    async def async_update(self):
+        """Only used by the generic entity update service."""
+        if not self.enabled:
+            return
+
+        await self.coordinator.async_request_refresh()
+
+    async def async_added_to_hass(self):
+        """Subscribe to updates."""
+        self.async_on_remove(
+            self.coordinator.async_add_listener(self.async_write_ha_state)
+        )
+
+
+class MeteoFranceRainSensor(MeteoFranceSensor):
+    """Representation of a Meteo-France rain sensor."""
+
+    @property
+    def state(self):
+        """Return the state."""
+        next_rain_date_locale = self.coordinator.data.next_rain_date_locale()
+        return (
+            dt_util.as_local(next_rain_date_locale) if next_rain_date_locale else None
+        )
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            ATTR_NEXT_RAIN_1_HOUR_FORECAST: [
+                {
+                    dt_util.as_local(
+                        self.coordinator.data.timestamp_to_locale_time(item["dt"])
+                    ).strftime("%H:%M"): item["desc"]
+                }
+                for item in self.coordinator.data.forecast
+            ],
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+        }
+
+
+class MeteoFranceAlertSensor(MeteoFranceSensor):
+    """Representation of a Meteo-France alert sensor."""
+
+    # pylint: disable=super-init-not-called
+    def __init__(self, sensor_type: str, coordinator: DataUpdateCoordinator):
+        """Initialize the Meteo-France sensor."""
+        self._type = sensor_type
+        self.coordinator = coordinator
+        dept_code = self.coordinator.data.domain_id
+        self._name = f"{dept_code} {SENSOR_TYPES[self._type][ENTITY_NAME]}"
+        self._unique_id = self._name
+
+    @property
+    def state(self):
+        """Return the state."""
+        return get_warning_text_status_from_indice_color(
+            self.coordinator.data.get_domain_max_color()
+        )
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            **readeable_phenomenoms_dict(self.coordinator.data.phenomenons_max_colors),
```

Comment:
Please set all the items explicitly instead. We want to know what they are and check the formatting.

Reply:
The `readeable_phenomenoms_dict ` code : 

https://github.com/hacf-fr/meteofrance-api/blob/a3ad71e73e4d75e1916d3b6a66322ccc4072b2ac/src/meteofrance/helpers.py#L61-L75

Is putting the content of the helper right ?

---

Is there a way to know the user language ? Because the helper handles it.

Or something in HA to translate attributes ?

Reply:
We don't support custom translations for state attributes only for the state. It's ok to keep values returned from the API for country specific integrations in the language of the country.